### PR TITLE
[release-3.9] oc rollout is unreliable for the error cases - use oc scale to ensure pods are stopped

### PR DESCRIPTION
**Pull request by richm**
The error cases were confusing oc rollout - so just simplify things
and use oc scale, and poll using oc get pod to make sure pods are stopped
and started.  I ran into a situation where the test was looking at the
pod log from the previous test and failing.
General clean up of test code, use artifact_out, etc.

### CODE REVIEW

File: test/curator.sh [hunk 1]
```diff
@@ -133,26 +142,27 @@ update_config_and_restart() {
...
+            --mount-path=/etc/curator --name=curator-config --overwrite 2>&1 | artifact_out
     fi
     restart_curator ${2:-}
 }
 
 if uses_config_maps ; then
-    origconfig=`mktemp`
-    os::log::debug "$( oc get configmap logging-curator -o yaml > $origconfig )" || :
+    origconfig=$( mktemp )
+    oc get configmap logging-curator -o yaml > $origconfig 2>&1 | artifact_out || :
```
Comment (nhosoi) [inline test/curator.sh]: `2>&1` would add output to stderr into $origconfig if any?  (or is it the purpose?)
I guess I'm worried too much... :)
```
oc get configmap logging-curator -o yaml > $origconfig 2| artifact_out || :
cat $origconfig | artifact_out || :
```

**Comment by richm** [comment]
many, many failures - flakage?
/retest

**Comment by richm** [comment]
curator logging seems to be working - worked for the journald case - not sure what's happening with the json-file case
/test json-file

**Comment by richm** [comment]
curator flake and remote-syslog flake
/retest

**Comment by richm** [comment]
remote-syslog flake
/test json-file

**Pull request merged**